### PR TITLE
MCLOUD-5752: Healthchecks for nginx and php fpm

### PR DESCRIPTION
**Pull request by VisanthSampath**
### Description

[MCLOUD-5752](https://jira.corp.magento.com/browse/MCLOUD-5752): Add container health checks for Nginx and PHP-FPM.

### Fixed Issues (if relevant)

1. magento/magento-cloud-docker#116: Add additional health checks.

### Manual testing scenarios

1. Build Nginx and PHP-FPM containers.
2. Run ./vendor/bin/ece-docker build:compose --mode="developer".
3. Run docker-compose up
4. Check if the Nginx and FPM containers are healthy.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

**Review by michaelcasey316 — COMMENTED** [review]
The code looks good. I have one question though, does `bin/ece-docker image:generate:php` need to be run to generate the appropriate configs for each of the PHP-FPM versions? For example, `images/php/7.1-fpm/etc/php-fpm.conf`, `images/php/7.2-fpm/etc/php-fpm.conf`, etc.

File: src/Compose/ProductionBuilder.php [hunk 1]
```diff
@@ -253,15 +253,29 @@ public function build(Config $config): Manager
 
         $manager->addService(
             self::SERVICE_FPM,
-            $this->serviceFactory->create(ServiceInterface::SERVICE_PHP_FPM, $phpVersion, ['volumes' => $volumesRo]),
+            $this->serviceFactory->create(
```
Comment (shiftedreality) [inline src/Compose/ProductionBuilder.php]: This should be part of the image, like ES

File: src/Compose/ProductionBuilder.php [hunk 2]
```diff
@@ -270,6 +284,12 @@ public function build(Config $config): Manager
                         'HTTPS_METHOD=noredirect',
                         'WITH_XDEBUG=' . (int)$config->hasServiceEnabled(ServiceInterface::SERVICE_FPM_XDEBUG)
                     ],
+                    self::SERVICE_HEALTHCHECK => [
```
Comment (shiftedreality) [inline src/Compose/ProductionBuilder.php]: This also should go to image

Comment (shiftedreality) [reply src/Compose/ProductionBuilder.php]: You can use `depends` like `build` or `deploy` container

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: src/Compose/ProductionBuilder.php (hunk 2 above)
Comment (VisanthSampath) [reply src/Compose/ProductionBuilder.php]: @shiftedreality The changes have been made. The health checks are now running inside the container.

**Comment by VisanthSampath** [comment]
> The code looks good. I have one question though, does `bin/ece-docker image:generate:php` need to be run to generate the appropriate configs for each of the PHP-FPM versions? For example, `images/php/7.1-fpm/etc/php-fpm.conf`, `images/php/7.2-fpm/etc/php-fpm.conf`, etc.

@michaelcasey316 The command has been run and committed those files under this pull request.

**Comment by michaelcasey316** [comment]
> > The code looks good. I have one question though, does `bin/ece-docker image:generate:php` need to be run to generate the appropriate configs for each of the PHP-FPM versions? For example, `images/php/7.1-fpm/etc/php-fpm.conf`, `images/php/7.2-fpm/etc/php-fpm.conf`, etc.
> 
> @michaelcasey316 The command has been run and committed those files under this pull request.

I'm looking at the files changed, and I don't see the versioned PHP images updated. Perhaps they were overwritten by one of your force pushes?

**Comment by VisanthSampath** [comment]
> > > The code looks good. I have one question though, does `bin/ece-docker image:generate:php` need to be run to generate the appropriate configs for each of the PHP-FPM versions? For example, `images/php/7.1-fpm/etc/php-fpm.conf`, `images/php/7.2-fpm/etc/php-fpm.conf`, etc.
> > 
> > 
> > @michaelcasey316 The command has been run and committed those files under this pull request.
> 
> I'm looking at the files changed, and I don't see the versioned PHP images updated. Perhaps they were overwritten by one of your force pushes?

@michaelcasey316 it was my bad.  They might have overwritten by force pushes. Added the versioned PHP-FPM files now.

**Review by michaelcasey316 — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Pull request merged**